lib/vanagon/component/dsl.rb

```diff
@@ -241,7 +241,10 @@ def install_configfile(source, target)
       # @param target [String] path to the desired symlink
       def link(source, target)
         @component.install << "#{@component.platform.install} -d '#{File.dirname(target)}'"
-        @component.install << "ln -s '#{source}' '#{target}'"
+        # Use a bash conditional to only create the link if it doesn't already point to the correct source.
+        # This allows rerunning the install step to be idempotent, rather than failing because the link
+        # already exists.
+        @component.install << "([[ `readlink '#{target}'` == '#{source}' ]] || ln -s '#{source}' '#{target}')"
```

Comment:
Suggestion: 

```bash
mckern@flexo Downloads $ ln -s ./IMG_0910.JPG somelink.jpg
mckern@flexo Downloads $ [[ somelink.jpg -ef somelink.jpg ]] && echo "true"
true
mckern@flexo Downloads $ [[ somelink.jpg -ef IMG_0910.JPG ]] && echo "true"
true
mckern@flexo Downloads $ [[ somelink.jpg -ef IMG_0911.JPG ]] && echo "true" || echo "false"
false
```

No subshells, no pipes, still a Bashism but portable to any shell if `[` is installed or it supports the extended test operator (`[[]]`).